axlearn/common/dit.py

```diff
@@ -211,15 +227,18 @@ def forward(self, input: Tensor) -> Tensor:
         """Generate the parameters for modulation.
 
         Args:
-            input: A tensor with shape [batch_size, ..., dim].
+            input: A tensor with shape [batch_size, dim] or [batch_size, num_length, dim].
 
         Returns:
             A list of tensors with length num_outputs.
-                Each tensor has shape [batch_size, ..., dim].
+                Each tensor has shape [batch_size, 1|num_length, dim].
         """
         cfg = self.config
         x = get_activation_fn(cfg.activation)(input)
         output = self.linear(x)
+        assert output.ndim in (2, 3)
```

Comment:
Raise a ValueError instead of `assert` (which should only be used to enforce internal logic errors and cannot be triggered by user error).

Reply:
As I clicked automerge, this your comment was not handled. I submitted #981 to handle it.